app/integration/project-generator/src/main/java/io/syndesis/integration/project/generator/ProjectGeneratorHelper.java

```diff
@@ -89,6 +90,11 @@ public static boolean filterDefaultDependencies(MavenGav gav) {
             answer = false;
         }
 
+        if ("org.apache.camel.k".equals(gav.getGroupId())) {
+            // Camel K does not work with s2i runtime
+            answer = false;
```

Comment:
not sure what this should do? Since knative is not working without camel-k (i.e. using old plain s2i) should we throw errors or maybe handle it at other levels?

Reply:
The S2I image builder takes all connectors and embed them into the image, but it fails with camel-k runtime libs because version is not in the standard s2i BOM. Instead of adding camel k to the s2i BOM, I thought it was better to exclude camel k libs from the S2I image (as you pointed out, they are not needed).

Reply:
I see, makes perfect sense.